### DISCUSSION TITLE
Anonymus report API

### DISCUSSION THREAD

**Discussion by fturibak** [ludeknovy/jtl-reporter#106] [Q&A]
Is there a possibility to generate Anonymous report using API.

Once I've run all Scenarios I would like to create link to report that could be send to e-mail addresses for additional verification.

It would be nice if there would be one link that could contain sub-links to Scenario reports.

P.S.
You gonna hate me for so much questions :-)

**Comment by ludeknovy** [comment]
Is that what you mean? 
https://jtlreporter.site/docs/introduction/features#anonymous-report-sharing

**Reply by ludeknovy** [reply]
https://user-images.githubusercontent.com/13610612/144429435-b7d26823-baf5-4905-b861-d329b28e91ac.mov

**Reply by fturibak** [reply]
Yes, but via API. 

I want to create it automatically after my test has finished for last run.

So I can send links to people responsible for particular test case.

**Reply by ludeknovy** [reply]
Ah, I think you should be able to use POST `/api/projects/:projectName/scenarios/:scenarioName/items/:itemId/share-tokens` with api token in headers.

**Reply by ludeknovy** [reply]
@fturibak what about sending the link automatically via email (using sendgrid or alike service) after test processing is done (currently it supports MS-TEAMS only)? Would that fit into your use case?

**Reply by fturibak** [reply]
This would be nice feature, but it would have to support sending mails via custom SMTP server..
